content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+}
+```
+
+h4(#rest-member-list). REST API
+
+The @Presence@ object exposes a "<span lang="default">@get@</span><span lang="csharp">@Get@</span>":#get method allowing a client to retrieve an array of all members currently present on the channel. In the REST client library this method directly queries "Ably's REST presence API":/api/rest-api#presence. No presence state is cached in the library itself, unlike in the "Realtime client library":/realtime.
+
+```[jsall]
+channel.presence.get(function(err, membersPage) {
+  console.log(membersPage.items.length + ' presence members in first page');
+  if(membersPage.hasNext()) {
+    membersPage.next(function(err, nextPage) { ... });
+  }
+});
+```
+
+```[ruby]
+members_page = channel.presence.get
+puts "#{members_page.items.length} presence members in first page"
+if members_page.has_next?
+  next_page = members_page.next
+end
+```
+
+```[python]
+members_page = channel.presence.get()
+print str(len(members_page.items)) + ' members present'
+if members_page.has_next():
+  next_page = members_page.next()
+```
+
+```[php]
+$membersPage = $channel->presence->get();
+echo(count($membersPage->items) . ' presence members in first page');
+if ($membersPage->hasNext()) {
+  $nextPage = $membersPage.next();
+}
+```
+
+```[java]
+PaginatedResult<PresenceMessage> membersPage = channel.presence.get(null);
+System.out.println(membersPage.items().length + " members in first page");
+if(membersPage.hasNext()) {
+  PaginatedResult<PresenceMessage> nextPage = membersPage.next();
+  System.out.println(nextPage.items().length + " members on 2nd page");
+}
+```
+
+```[csharp]
+  PaginatedResult<PresenceMessage> membersPage = await channel.Presence.GetAsync();
+  Console.WriteLine(membersPage.Items.Count + " members in first page");
+  if(membersPage.HasNext)
+  {
+    PaginatedResult<PresenceMessage> nextPage = await membersPage.NextAsync();
+    Console.WriteLine(nextPage.Items.Count + " members on 2nd page");
+  }
+```
+
+```[objc]
+[channel.presence get:^(ARTPaginatedResult<ARTPresenceMessage *> *membersPage, ARTErrorInfo *error) {
+  NSLog(@"%lu members in first page", [membersPage.items count]);
+  if (membersPage.hasNext) {
+    [membersPage next:^(ARTPaginatedResult<ARTPresenceMessage *> *nextPage, ARTErrorInfo *error) {
+      NSLog(@"%lu members on 2nd page", [nextPage.items count]);
+    }];
+  }
+}];
+```
+
+```[swift]
+channel.presence.get { membersPage, error in
+  let membersPage = membersPage!
+  print("\(membersPage.items.count) in first page")
+  if membersPage.hasNext {
+    membersPage.next { nextPage, error in
+      print("\(nextPage!.items.count) members on 2nd page")
+    }
+  }
+}
+```
+
+```[go]
+page, err := channel.Presence.Get(nil)
+fmt.Println("%d messages in first page\n", len(page.PresenceMessages()))
+if page.hasNext() {
+  page2, err := page.Next()
+  fmt.Println("%d messages on 2nd page!\n", len(page2.PresenceMessages()))
+}
+```
+
+
+h3(#presence-history). Presence History
```

Comment:
There seems to be some things missing before this section, according to the structure in Miro.

Reply:
We have a section "Presence history" here, but we don't explain what presence history is - there is no text at all in this h3 section. I would instead have some explanatory text, as per the Miro.